### PR TITLE
Implement responsive header and burger menu on small screens

### LINKED ISSUE #16: Make Header responsive
The logo needs and "Support Us" button fixing as it shrinks on a smaller view.

image

### PR DESCRIPTION
**Pull request by jromest**
## Description
Implement responsive header and burger menu on small screens

## Summary of Changes
- additional dependencies, install `body-scroll-lock`
- create shared breakpoints
- implement burger menu on a small screen in the header
- fix horizontal and vertical scroll in mobile 

## Screenshots
**Header**
Screen Shot 2020-10-13 at 6 37 17 PM

**Open Menu**
Screen Shot 2020-10-13 at 6 36 13 PM

Closes #16

### CODE REVIEW

**Comment by dorelljames** [comment]
Thank you for your contribution @jromest but I see one thing that needs to be fixed. I'm not sure if you noticed but the current deploy at https://deploy-preview-22--reactcebu-sitev2.netlify.app/ shows this previous issue we had on the site. See the styling on the newsletter form. 

image

Once that's fixed, I'd be happy to merge it. 😊

**Comment by jromest** [comment]
Hey @dorelljames, I think you forgot to load ng styled-components plugin in gatsby-config, adding it on the gatsby-config solves the issue. 😄

File: src/components/Header/Header.styles.ts [hunk 1]
```diff
@@ -1,36 +1,64 @@
...
+  justify-content: center;
+  flex-direction: column;
+  width: 100%;
+  height: 100vh;
+  position: fixed;
+  top: 0;
+  bottom: 0;
+  right: ${({ isOpen }) => (isOpen ? "0" : "-100%")};
+  transition: right 0.3s ease;
+  background-color: #fff;
```
Comment (dorelljames) [inline src/components/Header/Header.styles.ts:36]: Can you please add a `z-index` of at least a value of 5 to resolve this little overlap issue. :)

image

**Review by dorelljames — CHANGES_REQUESTED** [review]
Please add the necessary styling to finalize this PR. Thank you so much for your contribution @jromest. Well done. :)

**Review by siegblink — COMMENTED** [review]
I believe this is good to go although I have yet to test this in my local environment.
Bai @dorelljames your thoughts?

**Comment by dorelljames** [comment]
I agree with you @siegblink. Thanks. Will merge now. Thanks once again @jromest :)

**Pull request merged**